src/extensionApi.ts

```diff
@@ -81,6 +81,71 @@ export class CommandHandler {
         }
     }
 
+    public async debugServer(context?: Protocol.ServerState): Promise<Protocol.StartServerResponse> {
+
+        const debugInfo = await this.serversData.retrieveDebugInfo(context.server);
+
+        if (debugInfo && debugInfo.properties['debug.details.type'].indexOf('java') >= 0) {
+            if (vscode.extensions.getExtension('vscjava.vscode-java-debug') === undefined) {
+                vscode.window.showErrorMessage('Debugger for Java extension is required. Install/Enable it before proceeding.');
+                return;
+            }
+        } else {
+            vscode.window.showErrorMessage(`Vscode-Adapters doesn\'t support debugging with ${debugInfo.properties['debug.details.type']} language at this time.`);
+            return;
+        }
+
+        // list all projects that belong to the current workspace allowing the user choose among them or pick a new one
+        let optionsQuickPick: Array<{label: string, description: string}>  = [{ label: 'Add a new project', description: undefined}];
+        if (vscode.workspace.workspaceFolders !== undefined) {
+            optionsQuickPick = optionsQuickPick.concat(vscode.workspace.workspaceFolders.map(x => ({label: x.name, description: x.uri.toString()})));
+        }
+
+        let folderProject: vscode.Uri = await vscode.window.showQuickPick(optionsQuickPick,
+                                                { placeHolder: 'Select the project you want to debug', ignoreFocusOut: true }
+                                        ).then(folder => {
+                                            return (!folder || folder.description === undefined ? undefined : vscode.Uri.parse(folder.description));
+                                        });
+
+        if (!folderProject) {
+            folderProject = await vscode.window.showOpenDialog({
+                canSelectFiles: true,
+                canSelectMany: false,
+                canSelectFolders: true,
+                openLabel: 'Open Project to debug'
+            } as vscode.OpenDialogOptions).then(
+                folder => {
+                    if (folder != null && folder.length > 0) {
+                        vscode.workspace.updateWorkspaceFolders(vscode.workspace.workspaceFolders ? vscode.workspace.workspaceFolders.length : 0,
+                                                                null,
+                                                                { uri: vscode.Uri.parse(folder[0].path) });
+                        return folder[0];
+                    }
+                }
+            );
+        }
+
+        if (!folderProject) return;
+
+        return this.startServer('debug', context).then(
+            status => {
```

Comment:
don't we need to check `status` field ?

Reply:
Afaik the startServer will return a resolved promise if the server is started so it will get inside the "then" body only in that case. I didn't use a catch because if the promise is rejected is already managed in the executeCommand method. What do you think?

Reply:
No what is the server returns a status object that is containing an error indication ?

Reply:
I would expect that it has a status different from ok if it has an error indication inside. In the startServer method it's checked if (!StatusSeverity.isOk(response.status)) { . Is there any chance to receive an ok status with an error inside? I must check